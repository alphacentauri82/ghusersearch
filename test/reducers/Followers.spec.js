import followersReducer from 'reducers/Followers';

describe('Reducer: followers', () => {

  describe('when passed no state', () => {
    it('should return the initial state', () => {
      expect(
        followersReducer(undefined, {})
      ).toMatchSnapshot();
    });
  });

  describe('when a followers request is pending', () => {
    it('should set isPending to true', () => {
      const beforeState = {
        isPending: false,
      };
      const afterState = followersReducer(beforeState, {type: 'FOLLOWERS_REQUEST'});

      expect(
        afterState
      ).toMatchSnapshot();
    });
  });

  describe('when a followers request is successful', () => {
    it('should update the state with the repos', () => {
      const beforeState = {
        isPending: true,
        result: [],
      };
      const action = {
        payload: {
          result: [1, 2, 3],
        },
        type: 'FOLLOWERS_SUCCESS',
      };
      const afterState = followersReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toBe(beforeState);
    });
  });

  describe('when a search request fails', () => {
    it('should update the state with the error message', () => {
      const beforeState = {
        error: null,
        isPending: true,
      };
      const err = new Error('the server died');
      err.response = {foo: 'bar'};
      const action = {
        error: true,
        payload: err,
        type: 'FOLLOWERS_FAILURE',
      };
      const afterState = followersReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toBe(beforeState);
    });
  });

});
