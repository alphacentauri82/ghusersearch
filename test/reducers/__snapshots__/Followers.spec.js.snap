// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: followers when a followers request is pending should set isPending to true 1`] = `
Object {
  "isPending": true,
}
`;

exports[`Reducer: followers when a followers request is successful should update the state with the repos 1`] = `
Object {
  "error": null,
  "isPending": false,
  "result": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`Reducer: followers when a search request fails should update the state with the error message 1`] = `
Object {
  "error": [Error: the server died],
  "isPending": false,
}
`;

exports[`Reducer: followers when passed no state should return the initial state 1`] = `
Object {
  "error": null,
  "isPending": true,
  "result": Array [],
}
`;
